### ISSUE TITLE
Alpakka: JMS session isn't closed after ack timeout when stream is failed

### ISSUE THREAD

**Issue by matwojcik** [akka/alpakka#2039]
### Versions used 

Alpakka version: 1.1.2

### Behaviour

When consumer is defined to have ack timeout and fails the stream on ack timeout, then after timeout the underlying JMS session is not closed. It results in session being still open and brokers (like ActiveMQ) assign messages to this session, making other connection unable to read them.

### Reproducible Test Case

```scala
val jmsSource =
  JmsConsumer.txSource(
    JmsConsumerSettings(actorSystem.settings.config.getConfig(JmsConsumerSettings.configPath), connectionFactory)
      .withSessionCount(1)
      .withAckTimeout(1.second)
      .withQueue("test-queue")
      .withFailStreamOnAckTimeout(true)
  )

jmsSource
  .map { txEnvelope =>
    Thread.sleep(5000L)
    txEnvelope.commit()
    txEnvelope.message
  }
  .runWith(Sink.seq)
```

**Comment by ennru** [comment]
Thank you for reporting.
Would you be in a position to suggest the necessary changes in a Pull Request?

**Comment by matwojcik** [comment]
@ennru unfortunately I'm not sure where the bug origins from :/

**Comment by ennru** [comment]
Fixed with #2048

**Comment by kubukoz** [comment]
Thanks!
